### PR TITLE
Added smoke tests for Functions.

### PR DESCRIPTION
**Pull request by allisonbm92**
This commit adds two simple smoke tests for Functions.

### CODE REVIEW

File: smoke-tests/src/functions/java/com/google/firebase/testing/functions/FunctionsTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@RunWith(AndroidJUnit4.class)
+public final class FunctionsTest {
+
+  @Rule public final ActivityTestRule<Activity> activity = new ActivityTestRule<>(Activity.class);
+
+  @Test
+  public void callFakeFunctionShouldFail() throws Exception {
+    FirebaseFunctions functions = FirebaseFunctions.getInstance();
+    Task<HttpsCallableResult> task = functions.getHttpsCallable("clearlyFake31").call();
+    Tasks2.waitForFailure(task);
```
Comment (yifanyang) [inline smoke-tests/src/functions/java/com/google/firebase/testing/functions/FunctionsTest.java]: I assume there is no way to check error code for functions either?

File: smoke-tests/src/functions/java/com/google/firebase/testing/functions/FunctionsTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void callAddNumbersShouldReturnResult() throws Exception {
+    FirebaseFunctions functions = FirebaseFunctions.getInstance();
+    HashMap<String, Object> data = new HashMap<>();
+    data.put("firstNumber", 13);
+    data.put("secondNumber", 17);
+
+    Task<HttpsCallableResult> task = functions.getHttpsCallable("addNumbers").call(data);
```
Comment (yifanyang) [inline smoke-tests/src/functions/java/com/google/firebase/testing/functions/FunctionsTest.java]: Probably unrelated, but how come this method takes arbitrary string but not an enum or sort of thing?

Comment (allisonbm92) [reply smoke-tests/src/functions/java/com/google/firebase/testing/functions/FunctionsTest.java]: You can create functions with arbitrary names using the CLI.

File: smoke-tests/src/functions/java/com/google/firebase/testing/functions/FunctionsTest.java (hunk 1 above)
Comment (allisonbm92) [reply smoke-tests/src/functions/java/com/google/firebase/testing/functions/FunctionsTest.java]: Yeah.

**Pull request merged**